Yoast/wordpress-seo
# Tests/WP: use PSR-4

## Context

* Improve dev experience 

## Summary

This PR can be summarized in the following changelog entry:

* Tests: use PSR-4

## Relevant technical choices:

### Tests/WP: rename the test directory to WP

... and use ProperCase for all subdirectories to allow for supporting PSR-4.

Includes:
* Updating the names for the scripts to run these tests in the `composer.json` file.
    Note: the "old" `integration-test` and `integration-coverage` Composer scripts will still be available for the time being to facilitate the switch over and give the teams a little time to get used to the new script names.
* Renaming the `phpunit-integration.xml.dist` file to `phpunit-wp.xml.dist` and update it to match the other changes.
* Update the terminology used in the GH actions `test` workflow.
* Update the `.gitignore` file to match.
* Update the `phpcs.xml.dist` file to match.

### Tests/WP: namespace the test bootstrap file

Includes:
* Changing unqualified constants and function calls to fully qualified constants and function calls.
* Changing three named function declarations to closures.
* Making some minor tweaks to the PHPCS config, which can be removed once YoastCS 3.0 will be used.

### Tests/WP: namespace and rename the base test case

Includes:
* Removing prefixes from the test class name which are not needed in namespaced classes (`WPSEO/Yoast`) and where relevant, update references in other test files.
* Renaming the file to match the class name.
* Removing the file docblock.
* Adding the necessary import `use` statements.
* Changing unqualified constants and function calls to fully qualified constants and function calls.
* Adding import `use` statements to all files in which the class is _used_ and updating the class name where it is "used".

Includes removing the `framework` subdirectory and moving the base `TestCase` class to the root directory for these tests.

### Tests/WP: namespace and rename the WPSEO_Admin_Bar_Menu_Double

Includes:
* Removing prefixes from the test class names which are not needed in namespaced classes (`WPSEO/Yoast`) and where relevant, update references in other test files.
* Renaming the file to match the class name.
* Removing the file docblock.
* Adding the necessary import `use` statements.
* Adding import `use` statements to all files in which the class is _used_ and updating the class name where it is "used".

### Tests/WP: namespace and rename the `WPSEO_Admin_Editor_Specific_Repla…
### Tests/WP: namespace and rename the `WPSEO_Dismissible_Notification_Do…
### Tests/WP: namespace and rename the `WPSEO_Dismissible_Notification_Do…
### Tests/WP: namespace and rename the WPSEO_MyYoast_Api_Request_Double
### Tests/WP: namespace and rename the WPSEO_Sitemap_Image_Parser_Double
### Tests/WP: namespace and rename the WPSEO_Sitemaps_Renderer_Double
### Tests/WP: namespace and rename the WPSEO_Upgrade_Double
### Tests/WP: namespace and rename the WPSEO_Content_Images_Double
### Tests/WP: namespace and rename the `WPSEO_Gutenberg_Compatibility_Dou…
### Tests/WP: namespace and rename the WPSEO_Plugin_Availability_Double
### Tests/WP: namespace and rename the Post_Type_Sitemap_Provider_Double
### Tests/WP: namespace and rename the WPSEO_Sitemaps_Double
### Tests/WP: namespace and rename the WPSEO_Suggested_Plugins_Double
### Tests/WP: namespace and rename the Yoast_Form_Double
### Tests/WP: namespace and rename the Yoast_Notification_Center_Double
### Tests/WP: namespace and rename the Front_End_Integration_Double
### Tests/WP: namespace and rename the Statistics_Mock
### Tests/WP: namespace and rename the WPSEO_Capability_Manager_Double
### Tests/WP: namespace and rename the WPSEO_Meta_Columns_Double
### Tests/WP: namespace and rename the WPSEO_Primary_Term_Double

### Tests/WP: use PSR-4

* Move files into appropriate subdirectories
* Namespace all test files and remove file docblocks.
    Includes making all references to global functions and constants fully qualified and importing all used classes.
* Make sure the class name reflects the _class under test_.
* Remove prefixes from the test class names which are not needed in namespaced classes (`WPSEO/Yoast`) and where relevant, update references to those classes in the test files.
* Ensure all test class file names match the name of the test class.
* Change the Composer `autoload-dev` directive to use PSR-4 autoloading for all tests.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* _N/A_ If the tests pass, we're good.
